src/Uno.UWP/Devices/Sensors/ProximitySensor.Android.cs

```diff
@@ -0,0 +1,109 @@
+﻿#nullable enable
+
+using System;
+using System.Globalization;
+using System.Linq;
+using Android.Hardware;
+using Android.Runtime;
+using Uno.Devices.Enumeration.Internal;
+using Uno.Devices.Sensors.Helpers;
+
+namespace Windows.Devices.Sensors;
+
+public partial class ProximitySensor
+{
+	private Sensor? _sensor;
+	private ProximitySensorListener? _listener;
+
+	/// <summary>
+	/// Gets the device identifier.
+	/// </summary>
+	public string DeviceId { get; }
+
+	/// <summary>
+	/// Gets the maximum distance from the proximity sensor to the detected object.
+	/// </summary>
+	public uint? MaxDistanceInMillimeters => (uint)Math.Round(_sensor!.MaximumRange * 10);
+
+	/// <summary>
+	/// Obtains the proximity sensor from its identifier.
+	/// </summary>
+	/// <param name="sensorId">The sensor identifier.</param>
+	/// <returns>Returns the ProximitySensor object from its identifier.</returns>
+	public static ProximitySensor? FromId(string sensorId)
+	{
+		if (!DeviceIdentifier.TryParse(sensorId, out var sensorIdentifier))
+		{
+			return null;
+		}
+
+		var sensorManager = SensorHelpers.GetSensorManager();
+		var sensors = sensorManager.GetSensorList(Android.Hardware.SensorType.Proximity);
+		var androidSensor = sensors?.FirstOrDefault(s =>
+			sensorIdentifier.Id.Equals(
+				s.Id.ToString(CultureInfo.InvariantCulture), StringComparison.Ordinal));
+
+		var sensor = new ProximitySensor(sensorId);
+		sensor._sensor = androidSensor;
+		return sensor;
+	}
+
+	partial void StartReading()
+	{
+		_listener = new ProximitySensorListener(this);
+		SensorHelpers.GetSensorManager().RegisterListener(
+			_listener,
+			_sensor,
+			SensorDelay.Normal);
+	}
+
+	partial void StopReading()
+	{
+		if (_listener is not null)
+		{
+			SensorHelpers.GetSensorManager().UnregisterListener(_listener, _sensor);
+			_listener.Dispose();
+			_listener = null;
+		}
+	}
```

Comment:
Quite curious about thread-safety here.

Is it realistic that two threads call `StopReading`, and both go into the if condition, but one of them will set _listener to null earlier, then the other will throw NullReferenceException?

Reply:
Realized now that the implementation of StartStopEventWrapper is already thread safe.
 https://github.com/unoplatform/uno/blob/0bf7c32d04800952fa732298268a7a3f1c548c8b/src/Uno.UWP/Helpers/StartStopEventWrapper.cs#L59-L71